build/dependencies.yaml

```diff
@@ -85,32 +85,43 @@ dependencies:
 
   # Golang
   - name: "golang: upstream version"
-    version: 1.16.7
+    version: 1.17
     refPaths:
     - path: build/build-image/cross/VERSION
     - path: cluster/addons/fluentd-elasticsearch/es-image/Dockerfile
       match: 'FROM golang\:\d+.\d+(alpha|beta|rc)?\.?(\d+)?'
-    - path: test/images/Makefile
-      match: GOLANG_VERSION=\d+.\d+(alpha|beta|rc)?\.?\d+
     - path: staging/publishing/rules.yaml
       match: 'default-go-version\: \d+.\d+(alpha|beta|rc)?\.?(\d+)?'
-    - path: cluster/images/etcd/Makefile
-      match: 'GOLANG_VERSION\?=\d+.\d+(alpha|beta|rc)?\.?(\d+)?'
+    - path: test/images/Makefile
+      match: GOLANG_VERSION=\d+.\d+(alpha|beta|rc)?\.?\d+
 
-  # Golang pre-releases are denoted as `1.y<pre-release stage>.z`
+  # TODO(releng): Bump after:
+  #               1. go1.17 is released
+  #               2. initial go1.17 PR has merged
+  - name: "golang: minimum supported version"
+    version: 1.16.0
+    refPaths:
+    - path: hack/lib/golang.sh
+      match: minimum_go_version=go([0-9]+\.[0-9]+)
+
+  # Golang pre-releases are denoted as `1.y<pre-release stage>`
   # Example: go1.16rc1
   #
-  # This entry is a stub of the major and minor version to allow dependency
-  # checks to pass when building Kubernetes using a pre-release of Golang.
-  - name: "golang: <major>.<minor>"
-    version: 1.16
+  # This entry is a stub of the major version to allow dependency checks to
+  # pass when building Kubernetes using a pre-release of Golang.
+  - name: "golang: 1.<major>"
+    version: 1.17
     refPaths:
     - path: build/build-image/cross/VERSION
-    - path: hack/lib/golang.sh
-      match: minimum_go_version=go([0-9]+\.[0-9]+)
```

Comment:
restore this stanza, it will pass the canary CI jobs and fail the blocking ones. once we have green canaries, we'll switch the blocking ones to go1.17 at which point this will become the only PR that can merge

Reply:
and I guess update golang.sh to require go1.17 now